examples/clip/README.md

```diff
@@ -0,0 +1,227 @@
+
+# Pick the top 5 most similar labels for the image
+image_features /= image_features.norm(dim=-1, keepdim=True)
+text_features /= text_features.norm(dim=-1, keepdim=True)
+similarity = nn.Softmax(axis=-1)(100.0 * image_features @ text_features.T)
+values, indices = similarity[0].topk(5)
+
+# Print the result
+print("\nTop predictions:\n")
+index2label=[]
+with open('./cifar-100-binar/fine_label_names.txt','r') as f:
+	for line in f:
+		index2label.append(line.strip('\n'))
+for value, index in zip(values, indices):
+    print(f"{index2label[index]:>16s}: {100 * float(value):.2f}%")
+```
+
+The output will look like the following (the exact numbers may be slightly different depending on the compute device):
+
+```
+Top predictions:
+
+           snake: 65.31%
+          turtle: 12.29%
+    sweet_pepper: 3.83%
+          lizard: 1.88%
+       crocodile: 1.75%
+```
+
+Note that this example uses the `encode_image()` and `encode_text()` methods that return the encoded features of given inputs.
+
+
+### Linear-probe evaluation
+
+The example below uses [scikit-learn](https://scikit-learn.org/) to perform logistic regression on image features.
+
+```python
+import clip
+from mindspore import ops
+import mindspore.dataset as ds
+from download import download
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+from tqdm import tqdm
+
+# Load the model
+model, preprocess = clip.load("./ViT-B-32.ckpt", device="Ascend")
+
+# Load the dataset
+cifar100_url = "https://mindspore-website.obs.cn-north-4.myhuaweicloud.com/notebook/datasets/cifar-100-binary.tar.gz"
+download(cifar100_url, "./", kind="tar.gz", replace=True)
+cifar100_test_iter = ds.Cifar100Dataset("cifar-100-binary", usage="test", shuffle=False)
+cifar100_train_iter = ds.Cifar100Dataset("cifar-100-binary", usage="train", shuffle=False)
+cifar100_test_iter = cifar100_test_iter.map(preprocess, input_columns=["image"])
+cifar100_test_iter=cifar100_test_iter.batch(100)
+cifar100_train_iter = cifar100_train_iter.map(preprocess, input_columns=["image"])
+cifar100_train_iter=cifar100_train_iter.batch(100)
+
+def get_features(dataset):
+    all_features = []
+    all_labels = []
+    for images, _, labels in tqdm(dataset):
+        features = model.encode_image(images)
+        all_features.append(features)
+        all_labels.append(labels)
+
+    return ops.cat(all_features).asnumpy(), ops.cat(all_labels).asnumpy()
+
+
+# Calculate the image features
+train_features, train_labels = get_features(cifar100_train_iter)
+test_features, test_labels = get_features(cifar100_test_iter)
+
+# Perform logistic regression
+classifier = LogisticRegression(random_state=0, C=0.316, max_iter=1000, verbose=1)
+classifier.fit(train_features, train_labels)
+
+# Evaluate using the logistic regression classifier
+predictions = classifier.predict(test_features)
+accuracy = np.mean((test_labels == predictions).astype(float)) * 100.
+print(f"Accuracy = {accuracy:.3f}")
```

Comment:
Suggest to show the accuracy of linear-probe and zero-shot image classification, which will finally prove the precision and correctness of our implementation

Reply:
Compare with the original CLIP paper,  how is the zero-shot and linear-probe ACC?